src/main/kotlin/uk/gov/dluhc/printapi/database/entity/Certificate.kt

```diff
@@ -95,30 +100,97 @@ class Certificate(
         return this
     }
 
-    fun getCurrentPrintRequest(): PrintRequest {
-        printRequests.sortByDescending { it.requestDateTime }
-        return printRequests.first()
+    fun getPrintRequestsByStatus(printRequestStatus: Status) =
+        printRequests.filter { it.getCurrentStatus().status == printRequestStatus }
+
+    fun addPrintRequestToBatch(batchId: String) {
+        getPrintRequestsByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH).forEach {
+            it.addPrintRequestStatus(
+                PrintRequestStatus(
+                    status = Status.ASSIGNED_TO_BATCH,
+                    eventDateTime = Instant.now(),
+                    message = null
+                )
+            )
+            it.batchId = batchId
+        }
+        assignStatus()
```

Comment:
Do you think it will be better to add an `addPrintRequestStatus` method in this class, which adds the `PrintRequestStatus` to the relevant `PrintRequest` and also calls the `assignStatus` method? This way we won't need to care about updating the status of the certificate each time we add a new `PrintRequestStatus`.
<pre>
getPrintRequestsByStatus(Status.PENDING_ASSIGNMENT_TO_BATCH).forEach {
     <b>       addPrintRequestStatus( it, </b>
                PrintRequestStatus(
                    status = Status.ASSIGNED_TO_BATCH,
                    eventDateTime = Instant.now(),
                    message = null
                )
            )
            it.batchId = batchId
        }

  <b>  private fun addPrintRequestStatus(printRequest: PrintRequest, printRequestStatus: PrintRequestStatus) { </b>
        printRequest.addPrintRequestStatus(printRequestStatus)
        assignStatus()
    }
</pre>

Reply:
Refactored to have separate private method.